### ISSUE TITLE
salt.states.boto_elbv2 module needs updating to create / delete load balancers and listeners

### ISSUE THREAD

**Issue by hibbert** [saltstack/salt#44515]
### Setup
N/A

### Steps to Reproduce Issue

I was looking for a way to use salt to automate my AWS ELB setup. It seems that the full ELB state which creates load balancers etc is just for classic load balancers - https://docs.saltstack.com/en/develop/ref/states/all/salt.states.boto_elb.html. There is a new module for the new load balancers using boto3 elbv2 but it only supports creating target groups and registering targets and not the create_load_balancer and create_listener from boto3 elbv2, so unable to fully automate all current ELB types.

### Versions Report
# salt --versions-report
Salt Version:
           Salt: 2017.7.2

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 1.5
      docker-py: 1.10.6
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: 1.2.5
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.5 (default, Aug  4 2017, 00:39:18)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.3.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4

System Versions:
           dist: centos 7.4.1708 Core
         locale: UTF-8
        machine: x86_64
        release: 3.10.0-693.2.2.el7.x86_64
         system: Linux
        version: CentOS Linux 7.4.1708 Core

**Comment by gtmanfred** [comment]
@team-boto can yall take a look at this?

Thanks,
Daniel

**Comment by rallytime** [comment]
@saltstack/team-boto ^

**Comment by ryan-lane** [comment]
boto_elb supports VPC and classic ELBs. You need to use subnets, rather than availability_zones for this.

boto_elbv2 is ALBs, not ELBs. boto_elbv2 will also support NLBs soon, but will never support classis or VPC ELBs, as those are v1 ELBs. This is an annoying naming scheme by AWS.

**Comment by hibbert** [comment]
Yeah, boto_elb looks fine for a classic load balancer. I was looking to create an Application Load Balancer, but the boto_elbv2 saltstack state currently only creates the target group and registers instances, but there doesn't seem to be anything to create the load balancer and listeners to point at the target groups

**Comment by ryan-lane** [comment]
Ah. Gotcha. I think there's a couple companies working on elbv2 module right now (one of which is Lyft), which may be adding this functionality. Right now we're focused on NLBs, but I think a lot of the code overlaps between NLBs and ALBs.

**Comment by twellspring** [comment]
@ryan-lane was looking to restart my work on elbv2.   Is Lyft still working on NLBs?   Any ETA on a PR so I don't re-invent the wheel.

**Comment by Oloremo** [comment]
Same question as @twellspring - what is the status of the NLB implementation? 

@saltstack/team-boto ?

**Comment by gtmanfred** [comment]
@saltstack/team-boto also, @saltstack/team-triage

**Comment by anitakrueger** [comment]
Any idea where this is at please? I had automated all our classic ELB creation with the boto_elb state and am now stuck because we're moving to application load balancers and I can't create them or the listeners via Salt, but only the target groups.

**Comment by anitakrueger** [comment]
This is definitely not stale. No activity for 2 years, but the feature is still missing unfortunately.

**Comment by contherad** [comment]
Any update on this issue?

**Comment by anitakrueger** [comment]
@rallytime How do we get some traction on this? Any ideas? Thank you!

**Comment by cachedout** [comment]
@anitakrueger Hi Anita. Neither @rallytime (or myself) are with SaltStack anymore. I'll cc: @sagetherage who might be able to help route this to the proper person. Cheers. :)

**Comment by sagetherage** [comment]
@anitakrueger right now this feature isn't a priority for the Open Core team, but would look to @tkwilliams who is knowledgeable with this module or the community at large to submit a PR or ask for an architectural discussion with our Community Manager @cassandrafaris and key team members.

**Comment by anitakrueger** [comment]
@sagetherage thank you.

@cachedout Sorry, didn't realize you and @rallytime had left Saltstack. It's a bit difficult to know who to contact these days and who to tag on github issues. If only there was a Saltteam handle or something ;)

**Comment by sagetherage** [comment]
@anitakrueger you make a good point - there is a mention for the @saltstack/team-core but it may get lost there as well, we do have a community manager now so that should help, but for backlog items I would need to think about how we could make that workflow less about individuals. I open to suggestions in the meantime! :)

**Comment by cachedout** [comment]
@sagetherage FWIW, one of the hiccups with using @saltstack/team-core is that it is not discoverable from people outside the SaltStack organization. That means that normal features like GitHub autocomplete doesn't work and you need to specifically know what you're looking for in advance.

**Comment by sagetherage** [comment]
thx @cachedout yes, it has flaws for sure, a part from knowing the individual I am not sure on this ATM, but yes open to other ideas. We are working to update the Contributing.rst and docs, perhaps we need something specific there, but that is only a half-baked idea.
